internal/servers/starter/starter_test.go

```diff
@@ -0,0 +1,450 @@
+						Cert:    "./testdata/dummyServer.crt",
+						CA:      "./testdata/dummyCa.pem",
+						Key:     "./testdata/dummyServer.key",
+					},
+				}),
+			},
+			checkFunc: func(s Server, err error) error {
+				if err != nil {
+					return errors.Errorf("return an error: %v", err)
+				}
+
+				if s == nil {
+					return errors.New("server is nil")
+				}
+
+				return nil
+			},
+		},
+
+		{
+			name: "initialize is faild when tls.New returns error",
+			opts: []Option{
+				WithConfig(&config.Servers{
+					TLS: &config.TLS{
+						Enabled: true,
+					},
+				}),
+			},
+			checkFunc: func(s Server, err error) error {
+				if err == nil {
+					return errors.New("error is nil")
+				}
+
+				if s != nil {
+					return errors.New("server is not nil")
+				}
+
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			srv, err := New(tt.opts...)
+			if err := tt.checkFunc(srv, err); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestSetupAPIs(t *testing.T) {
+	type handler struct {
+		http.Handler
+	}
+
+	type args struct {
+		cfg *tls.Config
+	}
+
+	type field struct {
+		cfg  *config.Servers
+		rest func(cfg *config.Server) []server.Option
+		grpc func(cfg *config.Server) []server.Option
+		gql  func(cfg *config.Server) []server.Option
+	}
+
+	type test struct {
+		name      string
+		args      args
+		field     field
+		checkFunc func([]servers.Option, error) error
+	}
+
+	tests := []test{
+		func() test {
+			fn := func(srv *grpc.Server) {}
+
+			return test{
+				name: "returns options and nil",
+				args: args{
+					cfg: new(tls.Config),
+				},
+				field: field{
+					cfg: &config.Servers{
+						Servers: []*config.Server{
+							{
+								Mode: server.REST.String(),
+								Name: "rest_srv",
+								Host: "rest_srv_host",
+								Port: 8080,
+							},
+							{
+								Mode: server.GRPC.String(),
+								Name: "grpc_srv",
+								Host: "grpc_srv_host",
+								Port: 8081,
+							},
+							{
+								Mode: server.GQL.String(),
+								Name: "gql_srv",
+								Host: "gql_srv_host",
+								Port: 8082,
+							},
+						},
+					},
+					rest: func(cfg *config.Server) []server.Option {
+						return []server.Option{
+							server.WithHTTPHandler(new(handler)),
+						}
+					},
+					grpc: func(cfg *config.Server) []server.Option {
+						return []server.Option{
+							server.WithGRPCRegistFunc(fn),
+						}
+					},
+					gql: func(cfg *config.Server) []server.Option {
+						return []server.Option{
+							server.WithHTTPHandler(new(handler)),
+						}
+					},
+				},
+				checkFunc: func(opts []servers.Option, err error) error {
+					if err != nil {
+						return errors.Errorf("returns an error: %v", err)
+					}
+
+					if len(opts) != 3 {
+						return errors.Errorf("length of options is wrong. want: %v got: %v", 3, len(opts))
+					}
+
+					return nil
+				},
+			}
+		}(),
+
+		{
+			name: "returns nil options and error when setup of RESR server fails",
```

Comment:
- `REST` is right ?

Reply:
oh, Thank you for your feedback:bow:
I fixed typo:bow: